src/Rector/AbstractDrupalCoreRector.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+namespace DrupalRector\Rector;
+
+use DrupalRector\Contract\DrupalCoreRectorInterface;
+use PhpParser\Node;
+use Rector\Core\Rector\AbstractRector;
+use Rector\PostRector\Collector\NodesToAddCollector;
+
+abstract class AbstractDrupalCoreRector extends AbstractRector implements DrupalCoreRectorInterface
+{
+
+
+    /**
+     * @readonly
+     * @var \Rector\PostRector\Collector\NodesToAddCollector
+     */
+    private $nodesToAddCollector;
+
+    public function __construct(NodesToAddCollector $nodesToAddCollector)
+    {
+        $this->nodesToAddCollector = $nodesToAddCollector;
+    }
+
+    public function refactor(Node $node)
+    {
+        if (version_compare(\Drupal::VERSION, $this->getVersion(), '<')) {
+            return null;
+        }
+        $result = $this->doRefactor($node);
+        if ($result === null) {
+            return $result;
+        }
```

Comment:
We should move this into a compiler pass instead that unregister rules that do not apply to the current version of Drupal core.